onnx/defs/tensor/defs.cc

```diff
@@ -786,10 +794,22 @@ ONNX_OPERATOR_SET_SCHEMA(
                 int chunk_size = split_dim_value / num_outputs;
                 split.resize(num_outputs, chunk_size);
               } else { // tensor needs to be split unevenly
-                int chunk_size = (split_dim_value / num_outputs) + 1;
-                int last_chunk_size = split_dim_value - (chunk_size * (num_outputs - 1));
-                split.resize(num_outputs - 1, chunk_size);
-                split.push_back(last_chunk_size);
+                const auto mode = getAttribute(ctx, "mode", "torch");
+                int chunk_size = split_dim_value / num_outputs;
+                if (mode == "torch") {
+                  int reduced_dims = num_outputs * (chunk_size + 1) - split_dim_value;
+                  for (int i=0; i<num_outputs-reduced_dims; i++) {
+                    split.push_back(chunk_size+1);
+                  }
+                  while (split.size() <= num_outputs) {
+                    split.push_back(chunk_size);
+                  }
+                } else {
+                  int chunk_size = (split_dim_value / num_outputs) + 1;
```

Comment:
Actually, this doesn't always work. Eg., if we split 7 values into 5, then this will produce 4, 4, 4, 4, -1 which doesn't make sense. We should do 4, 3, 0, 0, 0, I think. We may need to change the description accordingly. In the other mode, we will do 2, 2, 1, 1, 1.